src/Common/src/Interop/Unix/libc/Interop.addrinfo.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
+
+using socklen_t = System.UInt32;
+
+internal static partial class Interop
+{
+    internal static partial class libc
+    {
+        public const int AI_NUMERICHOST = 0x0004; // Don't use name resolution.
+        public const int AI_NUMERICSERV = 0x0400; // Don't use name resolution.
+
+        public const int NI_NUMERICHOST = 1; // Don't try to look up hostname.
+
+        public const int EAI_BADFLAGS = -1; // Invalid value for `ai_flags' field.
+        public const int EAI_NONAME = -2;   // NAME or SERVICE is unknown.
```

Comment:
I think I saw you state that there would be a follow-up work item to move these defines into .OSX.cs / .Linux.cs / etc. files as needed in places where they're different.  Just want to make sure that's not lost, and an issue is opened to track it.  We may also just want to jump directly to the expected final solution of having this all in native shims.